feat(redis): add DB menu MTA-2004 (#696)
scaleway/docs-content: databases/databases-for-redis/concepts.mdx

---
meta:
  title: Databases for Redis™  - Concepts
  description: This page explains all the concepts related to Databases for Redis™ .
content:
  h1: Concepts
  paragraph: This page explains all the concepts related to Databases for Redis™ .
tags: advanced settings
categories:
  - database
---

<Concept opened>

## Acknowledgement flag (ACKs)  

> Are [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) flags used to indicate that a packet was received successfully.

Reviewer: I'm not sure about this. How does this refer to redis?

</Concept>

<Concept opened>

## Allowed IPs

Access control list (ACL) rules [define permissions for remote access to a Database Instance](/databases/databases-for-postgresql-and-mysql/how-to/manage-allowed-ip-addresses/). A rule consists of an IP address or an IP range.

</Concept>

<Concept opened>

## Cluster 

> A Redis™ cluster consists of a set of primary and secondary nodes. The cluster nodes use hash partitioning to split the keyspace into 16,384 key slots. Each master of the cluster is responsible for a subset of those slots. Each secondary node replicates the data of one of the primary ones.

Reviewer: ```suggestion
A Redis™ cluster consists of a set of primary and secondary nodes. The cluster nodes use hash partitioning to split the keyspace into 16,384 key slots. Each primary of the cluster is responsible for a subset of those slots. Each secondary node replicates the data of one of the primary ones.
```

</Concept>

<Concept opened>

## Cluster mode

> Each Managed Database for Redis™ cluster consists of three compute Instances hosting each a primary node and a secondary node for one of the other nodes. If one of the primary nodes fails, the remaining nodes hold a vote and elect one of the remaining secondary nodes as the new primary node. When the failing node rejoins the cluster, it automatically becomes a secondary node and begins to replicate the data of another primary node. 

Reviewer: ```suggestion
Each Managed Database for Redis™ cluster consists of three and up to six compute Instances hosting each a primary redis instance and a secondary instance for one of the other nodes. If one of the primary nodes fails, the remaining nodes hold a vote and elect one of the remaining secondary nodes as the new primary node. When the failing node rejoins the cluster, it automatically becomes a secondary node and begins to replicate the data of another primary node. 
```

</Concept>

<Concept>

## Endpoint

> A point of connection to a database. The endpoint is associated with an IPv4 address and a port, and contains the information of whether the endpoint is read-write or not.

Reviewer: ```suggestion
A point of connection to a database. The endpoint is associated with one or more public IPv4 address and a port or private IPv4 and port if you are using private networks
```

</Concept>

<Concept>

## High availability

> Is an option that allows you to create a secondary server, with an up-to-date replica of the database. If the primary server fails for any reason, the secondary can take over requests, reducing downtime.

Reviewer: ```suggestion
The High Availability is handled through Redis™ clustering feature. This is a 3 nodes multi master architecture. In case of downtime on 1 node of the cluster, a new node is able to take over the requests with a minimal downtime.
```

</Concept>

<Concept>

## Horizontal Scaling

Feature that allows you to increase the size of your Managed Database for Redis™  infrastructure by adding more nodes to your Database Instance to increase your number of endpoints and distribute cache. 

</Concept>

<Concept>

## In-memory Database 

A database management system primarily build upon memory. Data is managed and stored within the main memory of an Instance. This allows the optimization of request speed and the improvement of application performance. 

</Concept>

<Concept>

## Managed database

Compared to traditional database management, which requires customers to provision their infrastructure and resources to manage their databases, managed databases offer the user access to a Database Instance without setting up the hardware or configuring the software.

</Concept>

<Concept>

## Max Memory

Is a configuration directive used to set the maximum amount of memory Redis™ can use for a data set. You can set this value via the Scaleway Console or by configuring the `available_settings` via API.

</Concept>

<Concept>

> ## Region

Reviewer: Redis is a zoned product

tags: advanced settings
categories:
  - database
---

<Concept opened>

## Availability Zone 

An Availability Zone refers to the geographical location in which your instance will be created. The latency between multiple AZs of the same region is low, as they have a common network layer. Find below the list of availability zones by region:

- France - Paris (`fr-par`)
  `fr-par-1`
  `fr-par-2`
  `fr-par-3`
- The Netherlands - Amsterdam (`nl-ams`)
  `nl-ams-1`
  `nl-ams-2` 
- Poland - Warsaw (`pl-waw`)
  `pl-waw-1`

</Concept>

<Concept opened>

## Allowed IPs

Access control list (ACL) rules [define permissions for remote access to a Database Instance](/databases/databases-for-postgresql-and-mysql/how-to/manage-allowed-ip-addresses/). A rule consists of an IP address or an IP range.

</Concept>

<Concept opened>

## Cluster 

A Redis™ cluster consists of a set of primary and secondary nodes. The cluster nodes use hash partitioning to split the keyspace into 16,384 key slots. Each primary of the cluster is responsible for a subset of those slots. Each secondary node replicates the data of one of the primary ones.

</Concept>

<Concept opened>

## Cluster mode

> Each Managed Database for Redis™ cluster consists of three and up to six compute Instances hosting each a primary Redis™ Instance and a secondary Instance for one of the other nodes. If one of the primary nodes fails, the remaining nodes hold a vote and elect one of the remaining secondary nodes as the new primary node. When the failing node rejoins the cluster, it automatically becomes a secondary node and begins to replicate the data of another primary node. 

Reviewer 2: "three and up to six" ?

Author: ```suggestion
Each Managed Database for Redis™ cluster consists of minimum three and maximum six compute Instances hosting each a primary Redis™ Instance and a secondary Instance for one of the other nodes. If one of the primary nodes fails, the remaining nodes hold a vote and elect one of the remaining secondary nodes as the new primary node. When the failing node rejoins the cluster, it automatically becomes a secondary node and begins to replicate the data of another primary node. 
```

</Concept>

<Concept>

## Endpoint

> A point of connection to a database. The endpoint is associated with one or more public IPv4 addresses and a port or with a private IPv4 and port (if you use [Private Networks]).

Reviewer 2: missing a link

> A point of connection to a database. The endpoint is associated with one or more public IPv4 addresses and a port or with a private IPv4 and port (if you use [Private Networks]).

Reviewer 2: ```suggestion
A point of connection to a database. The endpoint is associated with one or more public IPv4 addresses and a port, or with a private IPv4 and a port (if you use [Private Networks]).
```

</Concept>

<Concept>

## High availability

> High Availability is handled through the Redis™ clustering feature. This is a 3 nodes multi-master architecture. In case of downtime on a node in the cluster, a new node is able to take over the requests with minimal downtime.

Reviewer 2: ```suggestion
High Availability is handled through the Redis™ clustering feature. This is a 3-node multi-master architecture. In case of downtime on a node in the cluster, a new node is able to take over the requests with minimal downtime.
```

</Concept>

<Concept>

## Horizontal Scaling

> Feature that allows you to increase the size of your Managed Database for Redis™  infrastructure by adding more nodes to your Database Instance to increase your number of endpoints and distribute cache. 

Reviewer 2: ```suggestion
A feature that allows you to increase the size of your Managed Database for Redis™  infrastructure by adding more nodes to your Database Instance to increase your number of endpoints and distribute cache. 
```

</Concept>

<Concept>

> ## In-memory Database 

Reviewer 2: ```suggestion
## In-memory database 
```

A database management system primarily build upon memory. Data is managed and stored within the main memory of an Instance. This allows the optimization of request speed and the improvement of application performance. 

</Concept>

<Concept>

## Managed database

> Compared to traditional database management, which requires customers to provision their infrastructure and resources to manage their databases, managed databases offer the user access to a Database Instance without setting up the hardware or configuring the software.

Reviewer 2: ```suggestion
Compared to traditional database management, which requires customers to provision their own infrastructure and resources to manage their databases, managed databases offer the user access to a Database Instance without them needing to set up the hardware or configure the software.
```

</Concept>

<Concept>

## Max Memory

> Is a configuration directive used to set the maximum amount of memory Redis™ can use for a data set. You can set this value via the Scaleway Console or by configuring the `available_settings` via API.

Reviewer 2: ```suggestion
A configuration directive used to set the maximum amount of memory Redis™ can use for a data set. You can set this value via the Scaleway console or by configuring the `available_settings` via API.
```

</Concept>

<Concept>

## TCP Keep alive 

> Packet that is sent by a device when the [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) connection on a routing device has been idle for too long. The packet is sent to the peer with only the [Acknowledgement flag (ACK)](#acknowledgement-flag-(acks)) turned on. 

Reviewer 2: ```suggestion
A packet that is sent by a device when the [TCP](https://en.wikipedia.org/wiki/Transmission_Control_Protocol) connection on a routing device has been idle for too long. The packet is sent to the peer with only the [Acknowledgement flag (ACK)](#acknowledgement-flag-(acks)) turned on. 
```

</Concept>

<Concept>

> ## Vertical Scaling  

Reviewer 2: ```suggestion
## Vertical scaling  
```

---
meta:
  title: Databases for Redis™  - Concepts
  description: This page explains all the concepts related to Databases for Redis™ .
content:
  h1: Concepts
  paragraph: This page explains all the concepts related to Databases for Redis™ .
tags: advanced settings
categories:
  - database
---

<Concept opened>

## Availability Zone 

> An Availability Zone refers to the geographical location in which your instance will be created. The latency between multiple AZs of the same region is low, as they have a common network layer. Find below the list of availability zones by region:

Reviewer 3: ```suggestion
An Availability Zone refers to the geographical location in which your Instance will be created. The latency between multiple AZs of the same region is low, as they have a common network layer. Find below the list of availability zones by region:
```

  `pl-waw-1`

</Concept>

<Concept opened>

## Allowed IPs

Access control list (ACL) rules [define permissions for remote access to a Database Instance](/databases/databases-for-postgresql-and-mysql/how-to/manage-allowed-ip-addresses/). A rule consists of an IP address or an IP range.

</Concept>

<Concept opened>

## Cluster 

A Redis™ cluster consists of a set of primary and secondary nodes. The cluster nodes use hash partitioning to split the keyspace into 16,384 key slots. Each primary of the cluster is responsible for a subset of those slots. Each secondary node replicates the data of one of the primary ones.

</Concept>

<Concept opened>

## Cluster mode

Each Databases for Redis™ cluster consists of minimum three and maximum six compute Instances hosting each a primary Redis™ Instance and a secondary Instance for one of the other nodes. If one of the primary nodes fails, the remaining nodes hold a vote and elect one of the remaining secondary nodes as the new primary node. When the failing node rejoins the cluster, it automatically becomes a secondary node and begins to replicate the data of another primary node. 

</Concept>

<Concept>

## Endpoint

> A point of connection to a database. The endpoint is associated with one or more public IPv4 addresses and a port, or with a private IPv4 and a port (if you use [Private Networks](/network/vpc/concepts/#private-network)).

Reviewer 3: ```suggestion
An endpoint is a point of connection to a database. The endpoint is associated with one or more public IPv4 addresses and a port, or with a private IPv4 and a port (if you use [Private Networks](/network/vpc/concepts/#private-network)).
```

</Concept>

<Concept>

## High availability

> High Availability is handled through the Redis™ clustering feature. This is a 3-node multi-master architecture. In case of downtime on a node in the cluster, a new node is able to take over the requests with minimal downtime.

Reviewer 4: ```suggestion
High Availability is handled through the Redis™ clustering feature. This is a 3-nodes multi-master architecture. In case of downtime on a node in the cluster, a new node is able to take over the requests with minimal downtime.
```

</Concept>

<Concept>

## Horizontal Scaling

A feature that allows you to increase the size of your Redis™ Database Instance infrastructure by adding more nodes to your Database Instance to increase your number of endpoints and distribute cache. 

</Concept>

<Concept>

## In-memory database 

> A database management system primarily build upon memory. Data is managed and stored within the main memory of an Instance. This allows the optimization of request speed and the improvement of application performance. 

Reviewer 3: ```suggestion
A database management system primarily builds upon memory. Data is managed and stored within the main memory of an Instance. This allows the optimization of request speed and the improvement of application performance. 
```
